Test/Integration/Page/MultiStorePagesTest.php

```diff
@@ -0,0 +1,148 @@
+        $this->pageRepository = $this->objectManager->get(PageRepositoryInterface::class);
+        $this->pageCollectionFactory = $this->objectManager->get(CollectionFactory::class);
+
+        $this->pagesIndexer->executeFull();
+        $this->algoliaHelper->waitLastTask();
+    }
+
+    /***
+     * @magentoDbIsolation disabled
+     *
+     * @throws ExceededRetriesException
+     * @throws AlgoliaException
+     */
+    public function testMultiStorePageIndices()
+    {
+        // Check that every store has the right number of pages
+        foreach ($this->storeManager->getStores() as $store) {
+            $this->assertNbOfRecordsPerStore(
+                $store->getCode(),
+                'pages',
+                $store->getCode() === 'fixture_second_store' ? // we excluded 2 pages on setupStore()
+                    $this->assertValues->expectedExcludePages :
+                    $this->assertValues->expectedPages
+            );
+        }
+
+        $defaultStore = $this->storeRepository->get('default');
+        $fixtureSecondStore = $this->storeRepository->get('fixture_second_store');
+
+        try {
+            $aboutUsPage = $this->loadPage(self::ABOUT_US_PAGE_ID);
+        } catch (\Exception $e) {
+            $this->markTestIncomplete('Page could not be found.');
+        }
+
+        // Setting the page only for default store
+        $aboutUsPage->setStores([$defaultStore->getId()]);
+        $this->pageRepository->save($aboutUsPage);
+
+        $this->pagesIndexer->execute([self::ABOUT_US_PAGE_ID]);
+        $this->algoliaHelper->waitLastTask();
+
+        $this->assertNbOfRecordsPerStore(
+            $defaultStore->getCode(),
+            'pages',
+            $this->assertValues->expectedPages
+        );
+
+        $this->assertNbOfRecordsPerStore(
+            $fixtureSecondStore->getCode(),
+            'pages',
+            $this->assertValues->expectedExcludePages - 1
+        );
+    }
+
+    /**
+     * Loads page by id.
+     *
+     * @param int $pageId
+     *
+     * @return PageInterface
+     * @throws LocalizedException
+     */
+    private function loadPage(int $pageId): PageInterface
+    {
+        return $this->pageRepository->getById($pageId);
+    }
+
+    protected function resetPage(PageInterface $page): void
+    {
+        $page->setStores([0]);
+        $this->pageRepository->save($page);
+    }
+
+    /**
+     * @param StoreInterface $store
+     * @param bool $enableInstantSearch
+     *
+     * @return void
+     * @throws AlgoliaException
+     * @throws LocalizedException
+     * @throws NoSuchEntityException
+     */
+    protected function setupStore(StoreInterface $store, bool $enableInstantSearch = false): void
+    {
+        // Exclude 2 pages on second store
+        $excludedPages = $store->getCode() === 'fixture_second_store' ?
+            [['attribute' => 'no-route'], ['attribute' => 'home']]:
+            [];
+
+        $this->setConfig(
```

Comment:
Do we need to tear this down? Should we have a `tearDownStore` method I wonder...? 🤔

Reply:
tearDown() triggers. At the end of the test all indices destroys

Reply:
No need to tear it down here I think since the 2 fixture stores are removed by the rollback script of the Magento fixtures.

But having  tearDownStore method on the parent class of the multi stores tests could make sense indeed :)